posydon/unit_tests/popsyn/test_star_formation_history.py

```diff
@@ -0,0 +1,1059 @@
+        from scipy.stats import norm
+        mean0, mean1 = (np.log10(madau.mean_metallicity(z)) 
+                  - model_dict['sigma']**2 * np.log(10) / 2)
+        print(mean0, mean1)
+        
+        expected1 = np.array(
+            # integral from 0.001 to 0.01; Z_min = lowest bin edge
+            [norm.cdf(np.log10(0.01), mean0, model_dict['sigma'])
+             - norm.cdf(np.log10(0.001), mean0, model_dict['sigma']),
+            # integral from 0.01 to 0.02
+             (norm.cdf(np.log10(0.02), mean0, model_dict['sigma'])
+              - norm.cdf(np.log10(0.01), mean0, model_dict['sigma'])),
+            # integral from 0.02 to 1
+             (norm.cdf(np.log10(1), mean0, model_dict['sigma'])
+              - norm.cdf(np.log10(0.02), mean0, model_dict['sigma']))],)
+        
+        expected2 = np.array(
+            # integral from 0.001 to 0.01;Z_min = lowest bin edge
+            [norm.cdf(np.log10(0.01), mean1, model_dict['sigma'])
+             - norm.cdf(np.log10(0.001), mean1, model_dict['sigma']),
+            # integral from 0.01 to 0.02
+             (norm.cdf(np.log10(0.02), mean1, model_dict['sigma'])
+              - norm.cdf(np.log10(0.01), mean1, model_dict['sigma'])),
+            # integral from 0.02 to 1
+             (norm.cdf(np.log10(1), mean1, model_dict['sigma'])
+              - norm.cdf(np.log10(0.02), mean1, model_dict['sigma']))],
+                      )
+        expected = np.array([expected1, expected2])
+        np.testing.assert_allclose(result, expected)
+        
+        # Change Z_min to 0 to include the rest of the lowest mets
+        model_dict = {"sigma": 0.5, "Z_max": 0.3, "Z_min": 0}
+        madau = self.ConcreteMadau(model_dict)
+        result = madau.fSFR(z, met_bins)
+        
+        expected1 = np.array(
+            [norm.cdf(np.log10(0.01), mean0, model_dict['sigma']),
+                (norm.cdf(np.log10(0.02), mean0, model_dict['sigma'])
+                - norm.cdf(np.log10(0.01), mean0, model_dict['sigma'])),
+                (norm.cdf(np.log10(0.3), mean0, model_dict['sigma'])
+                - norm.cdf(np.log10(0.02), mean0, model_dict['sigma']))],)
+        expected2 = np.array(
+            [norm.cdf(np.log10(0.01), mean1, model_dict['sigma']),
+                (norm.cdf(np.log10(0.02), mean1, model_dict['sigma'])
+                - norm.cdf(np.log10(0.01), mean1, model_dict['sigma'])),
+                (norm.cdf(np.log10(0.3), mean1, model_dict['sigma'])
+                - norm.cdf(np.log10(0.02), mean1, model_dict['sigma']))],)
+        expected = np.array([expected1, expected2])
+        np.testing.assert_allclose(result, expected)
+        
+        # Test with normalise
+        model_dict = {"sigma": 0.5, "Z_max": 0.3, "Z_min": 0, "normalise": True}
+        madau = self.ConcreteMadau(model_dict)
+        result = madau.fSFR(z, met_bins)
+        print(result)
+        np.testing.assert_allclose(np.sum(result, axis=1), np.ones(2))
```

Comment:
For consistency.
```suggestion
        expected = np.ones(2)
        np.testing.assert_allclose(np.sum(result, axis=1), expected)
```

May consider to infer the `2` from the result's shape.